codegens/curl/lib/index.js

```diff
@@ -15,14 +15,30 @@ self = module.exports = {
     options = sanitizeOptions(options, self.getOptions());
 
     var indent, trim, headersData, body, redirect, timeout, multiLine,
-      format, snippet, silent, url;
+      format, snippet, silent, url, shellType, lineContinuationCharacter, quoteType;
 
     redirect = options.followRedirect;
     timeout = options.requestTimeout;
     multiLine = options.multiLine;
     format = options.longFormat;
     trim = options.trimRequestBody;
     silent = options.silent;
+    shellType = options.shellType;
```

Comment:
We should still check if `lineContinuationCharacter` is present in the option object and choose the shell type accounting that in too. And if the `lineContinuationCharacter` is chosen as `^` we should use shell type `cmd.exe` This might cause problems for people who have already used this option in the app and switched to `^`, we will default them back to `sh`.